backend/eventloop/application_event_loop/application_event_runner_deployments.go

```diff
@@ -272,7 +271,36 @@ func (a applicationEventLoopRunner_Action) handleNewGitOpsDeplEvent(ctx context.
 		sourceRepoURL:        gitopsDeployment.Spec.Source.RepoURL,
 		sourcePath:           gitopsDeployment.Spec.Source.Path,
 		sourceTargetRevision: gitopsDeployment.Spec.Source.TargetRevision,
-		automated:            strings.EqualFold(gitopsDeployment.Spec.Type, managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated),
+		// syncOptions:       if non-empty, it gets updated below.
+		automated: strings.EqualFold(gitopsDeployment.Spec.Type, managedgitopsv1alpha1.GitOpsDeploymentSpecType_Automated),
+	}
+
+	if gitopsDeployment.Spec.SyncPolicy != nil && len(gitopsDeployment.Spec.SyncPolicy.SyncOptions) != 0 {
+		// syncOption = gitopsDeployment.Spec.SyncPolicy.SyncOptions
+		for _, syncOptionString := range gitopsDeployment.Spec.SyncPolicy.SyncOptions {
+			// Checks for each SyncOption goes in this for loop
+			var checkSyncOption bool
+			// 1. Check for SyncOption "CreateNamespace=true" / empty string
+			if syncOptionString == "" {
+				checkSyncOption = true
+			}
+			if syncOptionString == "- CreateNamespace=true" {
+				checkSyncOption = true
+			}
+			if syncOptionString == prunePropagationPolicy {
```

Comment:
The check for `PrunePropagationPolicy` is present here because the e2e test demanded an automated `GitopsDeployment` Type which meant `PrunePropagationPolicy` will be present in the syncOption and that is how I've added a check for the same so that an automated type does not get invalidated because of the new syncOption.